### DISCUSSION TITLE
Strange behavior in the "books" example

### DISCUSSION THREAD

**Discussion by jamolina2021** [csells/go_router#368] [Q&A]
Hi,

In the "books" sample application, after logging in, the url (http://localhost:64453/#/books/popular) is displayed.
but, as you will see in the screenshot, the **arrow to go to the previous url** (that is, the login url http://localhost:64453/#/signin) **is active**, although it is not possible to go back.
This is a bug? 
![book_error_1](https://user-images.githubusercontent.com/90254671/151586813-309f9a67-4862-4d4b-854c-2b10d65fc52f.png)

Thanks.

Regards,
Jose

**Comment by csells** [comment]
It is possible to go back according to the browser but then the redirection code kicks in. This is expected behavior.

**Reply by lulupointu** [reply]
Redirection is typically somewhere where wrapping the redirection in `Router.neglect` would be useful don't use think? Otherwise this kind of redirection breaks the browser back button (especially on mobile where user can only go back from 1).
The best would be to be able to specify whether or not to use `Router.neglect` but I actually think that in this case this would be a better default. 
What do you think ?

**Reply by csells** [reply]
You can set `routeNeglect` in the `GoRouter` constructor today to turn off browser history by default: https://gorouter.dev/web-history

**Reply by lulupointu** [reply]
This is great but I do feel that sometime you would want both. Redirection is a great example: I will often want to use web history but use `Router.neglect` when redirecting so that the user doesn't get stuck as explained above. Does that make sense ?

**Reply by csells** [reply]
Does the use of `Router.neglect` cover that case?

```dart
ElevatedButton(
  // turn off history tracking in the browser for this navigation
  onPressed: () => Router.neglect(context, () => context.go('/page2'))},
  ...
),
```

Or are you looking for something else?

**Reply by lulupointu** [reply]
Oh yes I didn't see that my bad, that's perfect 😉
